test-api/src/main/java/org/opennms/smoketest/OpenNMSSeleniumTestCase.java

```diff
@@ -264,6 +267,25 @@ protected void failed(final Throwable e, final Description description) {
             } else {
                 LOG.debug("Driver can't take screenshots.");
             }
+            try {
+                LOG.debug("Attempting to dump DOM.");
+                final String domText = m_driver.findElement(By.tagName("html")).getAttribute("innerHTML");
+                final String domFileName = "target" + File.separator + "contents" + File.separator + description.getClassName() + "." + testName + ".html";
```

Comment:
Ah yeah, I was avoiding newer stuff because `master` still requires 1.7, but the NIO stuff was introduced then, wasn't it?